src/main/java/gregtech/loaders/recipe/RecyclingRecipes.java

```diff
@@ -1,111 +1,400 @@
+                UnificationEntry entry = OreDictUnifier.getUnificationEntry(stack);
+                if (entry != null) { // should always be true
+                    OrePrefix prefix = entry.orePrefix;
+
+                    // These are the highest forms that a Material can have (for Ingot and Dust, respectively),
+                    // so simply split the stacks and continue.
+                    if (prefix == OrePrefix.block || prefix == OrePrefix.dust) {
+                        splitStacks(outputs, stack, entry);
+                    } else {
+                        // Attempt to split and to shrink the stack, and choose the option that creates the
+                        // "larger" single stack, in terms of raw material amount.
+                        List<Tuple<ItemStack, MaterialStack>> split = new ArrayList<>();
+                        List<Tuple<ItemStack, MaterialStack>> shrink = new ArrayList<>();
+                        splitStacks(split, stack, entry);
+                        shrinkStacks(shrink, stack, entry);
+
+                        if (split.get(0).getSecond().amount > shrink.get(0).getSecond().amount) {
+                            outputs.addAll(split);
+                        } else outputs.addAll(shrink);
+                    }
+                }
+            } else outputs.add(new Tuple<>(stack, ms));
+        }
+
+        // Sort the List by total material amount descending.
+        outputs.sort(Comparator.comparingLong(e -> -e.getSecond().amount));
+
+        // Cut out "duplicate" outputs.
+        // For example, if there are blocks of Steel and nuggets of Steel, and the nuggets
+        // are preventing some other output from occupying one of the final slots of the machine,
+        // cut the nuggets out to favor the newer item instead of having 2 slots occupied by Steel.
+        Map<MaterialStack, ItemStack> temp = new HashMap<>();
+        for (Tuple<ItemStack, MaterialStack> t : outputs) {
+            if (temp.containsKey(t.getSecond())) continue;
+            temp.put(t.getSecond(), t.getFirst());
+        }
+
+        // Not enough unique elements in the result list map.
+        if (temp.size() < maxOutputs) {
+            int limit = maxOutputs - temp.size(); // maximum number of outputs to append
+            temp.putAll(outputs.stream()
+                    .filter(t -> !temp.containsKey(t.getSecond()))
+                    .limit(limit)
+                    .collect(Collectors.toMap(Tuple::getSecond, Tuple::getFirst)));
+        }
+
+        return temp.entrySet().stream()
+                .sorted(Comparator.comparingLong(e -> -e.getKey().amount))
+                .limit(maxOutputs)
+                .map(Entry::getValue)
+                .collect(Collectors.toList());
+    }
+
+    private static void splitStacks(List<Tuple<ItemStack, MaterialStack>> list, ItemStack originalStack, UnificationEntry entry) {
+        int amount = originalStack.getCount();
+        while (amount > 64) {
```

Comment:
This *could* cause problems, but since we are pretty much only using a permutation of a Dust or Ingot, the stack size should be consistently 64